app/src/main/java/com/cortado/mafkir/ui/MainActivity.kt

```diff
@@ -0,0 +1,94 @@
+package com.cortado.mafkir.ui
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.provider.ContactsContract
+import android.util.Log
+import android.view.View
+import androidx.navigation.NavController
+import androidx.navigation.findNavController
+import com.cortado.mafkir.Constants.Companion.CONTACT_PICKER_RESULT
+import com.cortado.mafkir.R
+import com.cortado.mafkir.notifications.InteractionsService
+import com.cortado.mafkir.permissions.MafkirPermissionsValidator
+import dagger.android.support.DaggerAppCompatActivity
+import kotlinx.android.synthetic.main.activity_main.*
+import javax.inject.Inject
+
+
+class MainActivity : DaggerAppCompatActivity() {
+    @Inject
+    lateinit var mafkirPermissionsValidator: MafkirPermissionsValidator
+
+    private lateinit var navController: NavController
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        mafkirPermissionsValidator.validate(this)
+
+        startService(Intent(this, InteractionsService::class.java))
+
+        setContentView(R.layout.activity_main)
+
+        navController = findNavController(R.id.container)
+
+        fab.setOnClickListener {
+            doLaunchContactPicker(it)
+        }
+    }
+
+    private fun onFloatingClicked(contact: String) {
```

Comment:
bad name, specify that it's the add button

wait actually this isnt even that, it's called after the contact was chosen

Reply:
d